konduit-serving-pipeline/src/main/java/ai/konduit/serving/pipeline/impl/step/ml/ssd/regression/RegressionOutputRunner.java

```diff
@@ -0,0 +1,160 @@
+
+
+        NDArray regressionOutput = data.getNDArray(inputName);
+        if (regressionOutput.shape().length > 2) {
+            throw new UnsupportedOperationException("Invalid input to ClassifierOutputStep: only rank 1 or 2 inputs are available, got array with shape" + Arrays.toString(regressionOutput.shape()));
+
+        }
+        regressionOutput = FloatNDArrayToDouble(regressionOutput);
+
+
+        boolean batch = false;
+        if (regressionOutput.shape().length == 2 && regressionOutput.shape()[0] > 1) {
+            batch = true;
+        }
+
+        Map<String, Integer> outputNames = step.names();
+
+        if (outputNames == null || outputNames.isEmpty()) {
+            throw new UnsupportedOperationException("RegressionOutputStep names field was not provided or is null");
+        }
+
+
+        if (!batch) {
+            double[] regressionOutputArr = squeze(regressionOutput);
+            for (Map.Entry<String, Integer> entry : outputNames.entrySet())
+                data.put(entry.getKey(), regressionOutputArr[entry.getValue()]);
+
+        }
+
+        if (batch) {
+
+            int bS = (int) regressionOutput.shape()[0];
+            double[][] y = regressionOutput.getAs(double[][].class);
+
+            for (Map.Entry<String, Integer> entry : outputNames.entrySet()) {
+                List<Double> list = new ArrayList<Double>();
+                for (int i = 0; i < bS; i++) {
+                    list.add(y[i][entry.getValue()]);
+                }
+                data.putListDouble(entry.getKey(), list);
+
+            }
+
+
+        }
+
+
+        return data;
+    }
+
+
+    private static double[] squeze(NDArray arr) {
+
+        // we have [numClasses] array, so do not modify nothing
+        if (arr.shape().length == 1) {
+            return arr.getAs(double[].class);
+        }
+
+        // i.e we have [1, numClasses] array
+        if (arr.shape().length == 2 && arr.shape()[0] == 1) {
+            return arr.getAs(double[][].class)[0];
+        }
+
+        return null;
+
+    }
+
+    private static NDArray FloatNDArrayToDouble(NDArray ndarr) {
+        if (ndarr.type() == NDArrayType.FLOAT || ndarr.type() == NDArrayType.FLOAT16 || ndarr.type() == NDArrayType.BFLOAT16) {
+            float[][] farr = ndarr.getAs(float[][].class);
+            double[][] darr = new double[(int) ndarr.shape()[0]][(int) ndarr.shape()[1]];
+            for (int i = 0; i < farr.length; i++) {
+                for (int j = 0; j < farr[i].length; j++) {
+                    darr[i][j] = round(Double.valueOf(farr[i][j]),7);
+                }
+            }
+
+            return NDArray.create(darr);
+        }
+        return ndarr;
+    }
+
+    private static double round(double value, int places) {
+        if (places < 0) throw new IllegalArgumentException();
+
+        BigDecimal bd = new BigDecimal(Double.toString(value));
+        bd = bd.setScale(places, RoundingMode.HALF_UP);
+        return bd.doubleValue();
+    }
```

Comment:
Let's remove these 2 methods, make the ClassifierOutputStep equivalents `public static` and call those instead.
The whole rounding /BigDecimal thing is both unnecessary and slow too btw, casting float to double would be fine instead if we were to keep these methods.

we'll be able to remove those ones too once this is fixed: https://github.com/KonduitAI/konduit-serving/issues/389